### ISSUE TITLE
GPU delegate: BROADCAST_TO is rejected even at rank 4, so Tensor.expand (GQA repeat_kv) takes any transformer off the GPU

### ISSUE THREAD

**Issue by john-rocky** [google-ai-edge/LiteRT#9191]
## Summary

The `LITERT_CL` (ML Drift) GPU delegate rejects `BROADCAST_TO` outright, including at **rank 4**:

```
ERROR: Following operations are not supported by GPU delegate:
BROADCAST_TO: Operation is not supported.
```

Because `CompiledModel` requires full GPU residency, a single `BROADCAST_TO` anywhere in the graph fails compilation for the whole model rather than falling back for that one node.

This matters more than the op name suggests, because **`Tensor.expand` on a runtime tensor lowers to it**. The most common instance is grouped-query attention's `repeat_kv`:

```python
# transformers-style repeat_kv — one line, and the graph is off the GPU
hidden = hidden[:, :, None, :, :].expand(b, n_kv, n_rep, s, d)
```

so every GQA transformer (Qwen3 / Llama / Gemma family text encoders and LMs) hits it when exported through `litert-torch` and compiled for GPU.

## Minimal repro

A single-op graph at rank 4, with no >4D tensor involved, so this is cleanly separated from the known >4D limitation:

```python
import torch, torch.nn as nn

class BroadcastToRank4(nn.Module):
    # x[1,4,1,64] -> [1,4,8,64]: one BROADCAST_TO, never leaves rank 4
    def forward(self, x):
        return x.expand(-1, -1, 8, -1) + 0.0

# exported with litert-torch, fp32, input [1,4,1,64]
```

Compiling the resulting `.tflite` with `CompiledModel` and `Accelerator.GPU` fails with the message above; the same graph runs fine on CPU.

## Versions

Verified on a Pixel 8a (Mali-G610, `LITERT_CL`, `CompiledModel`, default fp16 GPU compute):

| litert | rank-4 `BROADCAST_TO` probe |
|---|---|
| 2.1.3 | reject |
| 2.1.5 | reject |
| 2.1.6 | reject |

So this is not a regression — it has been unsupported across all three releases, and 2.1.6 is the current one.

## Workaround (exact, if it helps others)

Replacing the broadcast with an explicit concatenation keeps the head order and the numerics identical, and lowers to `CONCATENATION`:

```python
hidden = torch.cat([x[:, i:i+1].expand(b, n_rep, s, d) for i in range(n_kv)], dim=1)
```

That restores full GPU residency for the models above.

## Note on the earlier report

[#6851](https://github.com/google-ai-edge/LiteRT/issues/6851) covered `FILL` and `BROADCAST_TO` not being registered and was closed by the inactivity bot rather than by a fix, so I am opening this per the bot's suggestion. The rank-4-only probe and the `expand`/`repeat_kv` connection are new here — I thought that framing might be useful, since it means the op affects a whole class of transformer exports rather than only mask-creation patterns.

Happy to run any further probe on the device if that would help narrow it down.

## Environment

- Device: Pixel 8a, Mali-G610, `LITERT_CL` accelerator, `CompiledModel` API
- litert 2.1.3 / 2.1.5 / 2.1.6 (pip `ai-edge-litert`), same behavior on each
- Graph exported with `litert-torch`, fp32

**Comment by shivvaam0001** [comment]
Hi @john-rocky,

Thank you for bringing this to our attention.
Currently, the `BROADCAST_TO` operator is not listed among the officially supported operations for the [LiteRT GPU delegate](https://developers.google.com/edge/litert/performance/gpu).

Thanks!

**Comment by john-rocky** [comment]
Thank you for checking — agreed, the documented op list and the behavior match, and the feature framing is fine. Two small additions for the record:

- The classic GPU delegate declines it as well, not only `LITERT_CL`: a single-op rank-4 `BROADCAST_TO` graph (`[1,1,64,64] → [1,8,64,64]`, exported with litert-torch 0.9.2) run today on the same Pixel 8a via `benchmark_model --use_gpu=true` gives

  ```
  ERROR: Following operations are not supported by GPU delegate:
  BROADCAST_TO: Operation is not supported.
  No operations will run on the GPU, and all 1 operations will run on the CPU.
  ```

  so there is currently no GPU route for the stock `expand` lowering on either runtime.

- For roadmap weighting, a delegate kernel may not even be necessary: the concat rewrite in the issue body is numerically identical and fully GPU-resident, so a converter-side rewrite of `BROADCAST_TO` into `CONCATENATION` (or folding it into the consumer op) would unblock the GQA family without touching the delegates.

Happy to verify any candidate — delegate-side or converter-side — on this device.

**Comment by john-rocky** [comment]
Re-tested on ai-edge-litert 2.2.0 (yesterday's PyPI release, the first since this was filed): the rank-4 BROADCAST_TO rejection is unchanged from 2.1.6.

The single-op repro from my earlier comment ([1,1,64,64] → [1,8,64,64]) prints the identical message on both versions — "BROADCAST_TO: Operation is not supported. No operations will run on the GPU." It also reproduces on macOS through the Python CompiledModel API (Metal), so this isn't specific to the Android CL path.

Happy to re-run it against any candidate build — it takes a few seconds anywhere the wheel installs.
